src/lib/geoip/client.ts

```diff
@@ -87,11 +101,38 @@ export default class GeoipClient {
 		return false;
 	}
 
+	private matchNetwork(best: LocationInfo, sources: LocationInfoWithProvider[]): NetworkInfo | undefined {
+		if (best.asn && best.network) {
+			return {
+				asn: best.asn,
+				network: best.network,
+			};
+		}
+
+		const maxmind = sources.find(s => s.provider === 'maxmind');
+		if (maxmind?.asn && maxmind?.network) {
+			return {
+				asn: maxmind.asn,
+				network: maxmind.network,
+			};
+		}
```

Comment:
Only if city for maxmind is the same as best match. If cities are different, we don't want to mix them and return undefined.

Reply:
done